server/image_remove.go

```diff
@@ -31,14 +31,27 @@ func (s *Server) removeImage(ctx context.Context, imageRef string) error {
 	ctx, span := log.StartSpan(ctx)
 	defer span.End()
 
-	images, err := s.StorageImageServer().ResolveNames(s.config.SystemContext, imageRef)
+	// FIXME: The CRI API definition says
+	//      This call is idempotent, and must not return an error if the image has
+	//      already been removed.
+	// and this code doesn’t seem to conform to that.
+
+	// Actually Kubelet is only ever calling this with full image IDs.
+	// So we don't really need to accept ID prefixes nor short names;
+	// or is there another user?!
```

Comment:
cri-tools does an `ImageStatus` before removing the specified image, which leads into a ResolveNames call. So I guess we're fine.